base/statistics.jl

```diff
@@ -139,11 +176,8 @@ Like [`median`](@ref), but may overwrite the input vector.
 """
 function median!(v::AbstractVector)
     isempty(v) && throw(ArgumentError("median of an empty array is undefined, $(repr(v))"))
-    if eltype(v)<:AbstractFloat
-        @inbounds for x in v
-            isnan(x) && return x
-        end
-    end
+    eltype(v)>:Missing && any(ismissing, v) && return missing
+    eltype(v)>:AbstractFloat && any(isnan, v) && return NaN
```

Comment:
I broke this check by mistake but interestingly the tests didn't catch it because they only covered cases like `[1.0, NaN]`, which return `NaN` without it. The correct check seems to be `eltype(v)<:AbstractFloat || eltype(v)>:AbstractFloat`, right?

Ideally, we wouldn't need these explicit type checks (https://github.com/JuliaLang/julia/issues/21256).

Reply:
What if the input is say `Any[1, missing, 1]` or maybe even `Any[1, missing, NaN, 1]`?

Reply:
Since `Any >: Missing`, these return `missing`, as documented in the docstring (below).

Reply:
Regarding the other point, do you think `eltype(v)<:AbstractFloat || eltype(v)>:AbstractFloat` is OK here?